hazelcast/src/main/java/com/hazelcast/query/impl/IndexUtils.java

```diff
@@ -333,11 +351,31 @@ public static IndexConfig getIndexConfigFromXml(Node indexNode, boolean domLevel
                 res.getBitmapIndexOptions().setUniqueKey(uniqueKey);
                 res.getBitmapIndexOptions().setUniqueKeyTransformation(uniqueKeyTransformation);
             }
+        } else if (type == IndexType.SORTED) {
+            Node optionsNode = childElementWithName(indexNode, "btree-index", strict);
```

Comment:
I see that if a user uses this new config with other index types, we simply ignore it. Would it be better to check this and fail-fast for invalid configurations? I am not sure which behavior is better, so don't hesitate to ignore this comment if you think the current behavior is better

Reply:
IIRC we always ignore config that is not relevant